<a href="https://colab.research.google.com/github/erikalira/introducao-a-ciencia-de-dados/blob/main/Exerccio___Medidas_de_Centralidade_e_Disperso.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Introdução à Ciência de Dados - UFPB
Professor: Yuri Malheiros

## Medidas de Centralidade e Dispersão

A partir do dataset `movies.csv` calcule a média, mediana e desvio padrão para as colunas:

In [1]:
import pandas as pd
import math

movies = pd.read_csv("movies.csv")

In [2]:
def statistics (col) :
  print(f"media: {movies[col].mean()}")
  print(f"mediana: {movies[col].median()}")

  print(f'\ndiferença entre media e mediana: {math.fabs(movies[col].mean() - movies[col].median())}')
  print(f"desvio padrão: {movies[col].std()}")

### US Gross

In [3]:
statistics('US Gross')

media: 44002085.163744524
mediana: 22019465.5

diferença entre media e mediana: 21982619.663744524
desvio padrão: 62555311.39066204


### Production Budget

In [4]:
statistics('Production Budget')

media: 31069171.4484375
mediana: 20000000.0

diferença entre media e mediana: 11069171.4484375
desvio padrão: 35585913.44464446


### Running Time

In [5]:
statistics('Running Time (min)')

media: 110.19354838709677
mediana: 107.0

diferença entre media e mediana: 3.1935483870967687
desvio padrão: 20.171013677535424


### IMDB Rating

In [6]:
statistics('IMDB Rating')

media: 6.283467202141896
mediana: 6.4

diferença entre media e mediana: 0.11653279785810433
desvio padrão: 1.2522899386004838


### IMDB Votes

In [7]:
statistics('IMDB Votes')

media: 29908.64457831325
mediana: 15106.0

diferença entre media e mediana: 14802.644578313251
desvio padrão: 44937.58233504965


### Análise dos Resultados

Quais colunas tiveram valores muito diferentes entre média e mediana? Quais os seus desvios padrão?

r: US Gross, Production Budget e IMDB Votes. 62555311.39066204, 11069171.4484375 e 14802.644578313251, respectivamente.

Que conclusões podemos tirar desses resultados?

r: para as colunas com altos valores de desvio padrão, existe alta dispersão. Ou seja, existem valores muito distintos. O que tornam a media uma metrica não muito boa nesses casos.